Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -740,3 +756,26 @@ public enum RestError: Error, CustomStringConvertible {
         }
     }
 }
+
+// URL Session extension
+extension RestRequest: URLSessionDelegate {
+    
+    /// URL session function to allow trusting certain URLs
+    public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        let method = challenge.protectionSpace.authenticationMethod
+        let host = challenge.protectionSpace.host
+        switch (method, host) {
+        case (NSURLAuthenticationMethodServerTrust, self.url):
+            #if MAC_OS_X_VERSION_10_6
+                let trust = challenge.protectionSpace.serverTrust
+                let credential = URLCredential(trust: trust!)
```

Comment:
@ShihabMehboob I am not very familiar with this API so bear with me. I am looking at like 771 and I see we are  force unwrapping the `SecTrust ` variable returned when accessing the `serverTrust` field: https://developer.apple.com/documentation/foundation/urlauthenticationchallenge/1410012-protectionspace

If in the code above, we have the possibility of `trust` being nil, then we should not force unwrap it. That would cause a runtime error and a crash. Can you double check whether nil could be returned in our case?